### DISCUSSION TITLE
How can I inherit from BaseModel while keeping autocomplete functionality?

### DISCUSSION THREAD

**Discussion by RamiAwar** [pydantic/pydantic#2658] [Question] [answered]
Hi there,

I'm trying to create a NewBaseModel(BaseModel), but when I try to instantiate new instances of NewBaseModel, the type checking magical dust that applies to the __init__ in BaseModel is not transferred to NewBaseModel.

Can someone point me in the right direction to fix this?

**Comment by RamiAwar** [comment, marked as answer]
Turns out even BaseModel doesn't have the fields as parts of the constructor, my bad.

Autocomplete still works, but it's the constructor hints that don't for a base pydantic model.

**Comment by tl24** [comment]
It also depends on your editor.  PyCharm does autocomplete for Pydantic (possibly a plugin is needed), but VSCode does not.
